### ISSUE TITLE
Trigger on notification

### ISSUE THREAD

**Issue by BladeBTW** [antero111/plugin-presets#36]
Hey, back again.
A cool feature to the already very cool plugin could be a preset trigger based on the notification system in RuneLite.

Example usecase:
![image](https://user-images.githubusercontent.com/52164618/196004196-7cc56424-40bf-4414-af58-dfcb5d5e17ab.png)
In this screenshot I am mining in f2p on a few accounts, when an account is afk the screen turns blue (this is the notification setting > solid until cancelled in RL).

If I was able to trigger preset settings based on notifications, I could for example make player indicators under my character red when afk (to reduce eye strain).
Like in this picture:
![image](https://user-images.githubusercontent.com/52164618/196004258-c33f8237-8642-47de-8481-3e61c28a2a79.png)

I think it should be a button/icon (maybe an exclamation mark?) Or the icon windows uses for notifications:
![image](https://user-images.githubusercontent.com/52164618/196004279-9e32b8a3-185d-41a7-9ede-d2dbf0558770.png)

Wonder what you think. :)

**Comment by antero111** [comment]
If we were to add preset loading based on afk, then we will have someone asking for toggling presets when fishing, questing, when in combat or based on location (as in https://github.com/antero111/plugin-presets/issues/35).

I would love to add that kind of possibility but I think that at some point we have to discuss with the RL admins about if there is some limitations on this before starting to develop.

About your issue that the flashing light is strain in the eyes, maybe RL could have some settings to make it more modest like flashing only the borders (or make your pc:s tile red as you described). I'd suggest to ask in the RL Discord about this kind of feature or maybe create a discussion on the RL repo.

TLDR: If we decide to add more ways to toggle presets, something like this will be possible but otherwise I'd suggest that you asked about this "more modest flashing" in the RL Discord/Github.

**Comment by BladeBTW** [comment]
I get what you are saying, fair enough!

I can change the notification intensity by changing the opacity in RuneLite as is.

The option to trigger on notification just gives you a little bit of extra creative freedom with building your setup with the presets plugin.

As far as I know there is only 1 notification type RL can send. Which is why I thought would be a useful trigger for the plugin (and not complex).

You might me right though it might complicate the plugin too much.

Thank you for your response! 👍

**Comment by BladeBTW** [comment]
> 

I think especially in combination with the change preset on window focus it could be strong. You can have your client settings in 3 modes "unfocused/unfocused & notification (notification overwrites unfocus)/focused (focus will always overwrite notification).

**Comment by antero111** [comment]
I have to look how notifications work since if there is some universal notfication event it maybe could be implemented somewhat well.

**Comment by BladeBTW** [comment]
I've not developed much myself, but before the 3rd party client ban I had my own fork of RL. And I made custom color notifications per activity, when I was messing with that I believe it was the notifier.java file. And it seems to be universal.
But no rush, I think about your plugin a lot, and this was a thought I had.. lol :)

**Comment by antero111** [comment]
Hi Blade, 
you might be intrested in these issues about screen flashing:

https://github.com/runelite/runelite/issues/11679
https://github.com/runelite/runelite/issues/12147

Maybe you can get some improvements to happen from there for reducing eye strain with multiple flashing clients.

As for implementing notification preset activation, I have no timeline when I have time to implement such thing.

**Comment by BladeBTW** [comment]
Thank you so much, I will be keeping an eye on these.

I've been using the functionality of the settings syncing from your presets
plugin daily. And it's made my gameplay a lot easier. Thanks again, from
the bottom of my heart. <3

**Comment by antero111** [comment]
I'll close this as not planned for now and leave the more modest notification feature to be implemented for example with runelite-watchdog or in the main client. 

Notifications are not particularly easy to work with as preset triggers and I think at the current state, would be really miniscule improvement on plugin presets. 

If we get to a point where presets would be activated by some game event or region id then we can have a think about this one.
